geode-connectors/src/main/java/org/apache/geode/connectors/jdbc/internal/TableKeyColumnManager.java

```diff
@@ -37,15 +39,15 @@ public String getKeyColumnName(Connection connection, String tableName) {
   }
 
   private String computeKeyColumnName(Connection connection, String tableName) {
```

Comment:
why not throw SQLException? I looked at the callers and they seem to all funnel back through SQLHandler read and write

Reply:
the method getKeyColumnName calls this private method in a computeIfAbsent call. Since that is a lambda, only a runtime exception can be thrown from it. To throw an SQLException, I would have to catch it in the computeIfAbsent , throw a new runtime exception, then catch that runtime exception and throw the inner SQLException again. That seems pretty ugly. We could undo using a computeIfAbsent call to get rid of the runtime exception requirement.